core/src/main/java/io/questdb/cairo/mv/MatViewDefinition.java

```diff
@@ -338,10 +354,17 @@ public void setPeriodSampler(TimestampSampler periodSampler) {
         this.periodSampler = periodSampler;
     }
 
+
+    public void updateBaseTableTimestampType(int baseTableTimestampType) {
+        this.baseTableTimestampType = baseTableTimestampType;
```

Comment:
is this tested? sampler has to be aligned to the type of the timestamp, which isn't being updated here. I suspect test, if exists, is deficient

Reply:
Good spot!